core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -88,6 +89,12 @@
    * index of segmentProperties in the segmentProperties holder
    */
   protected int segmentPropertiesIndex;
+
+  /**
+   * carbon table object
+   */
+  protected CarbonTable carbonTable;
```

Comment:
I am not sure why it is required

Reply:
carbonTable is required in prune method and isScanRequired method